assets/js/admin/course-theme-sidebar.js

```diff
@@ -3,49 +3,147 @@
  */
 import useCourseMeta from '../../react-hooks/use-course-meta';
+import { name as courseOutlineBlockName } from '../../blocks/course-outline/outline-block/block.json';
+import { name as courseModuleBlockName } from '../../blocks/course-outline/module-block/block.json';
+import { name as courseLessonBlockName } from '../../blocks/course-outline/lesson-block/block.json';
 
 const SENSEI_THEME = 'sensei-theme';
 const WORDPRESS_THEME = 'wordpress-theme';
+const SENSEI_PREVIEW_QUERY = 'sensei_theme_preview';
+
+const canPreview = ( block ) =>
+	block.name === courseLessonBlockName &&
+	block.attributes.id &&
+	! block.attributes.draft;
 
 /**
  * Course Theme Sidebar component.
  */
 const CourseThemeSidebar = () => {
 	const [ theme, setTheme ] = useCourseMeta( '_course_theme' );
 
+	const globalLearningModeEnabled =
+		window.senseiSettings.sensei_learning_mode_all || false;
+	const currentPost = useSelect( ( select ) =>
+		select( 'core/editor' ).getCurrentPost()
+	);
+
+	const firstLesson = useSelect( ( select ) => {
+		const { getBlocks, getBlockAttributes } = select( 'core/block-editor' );
+		const blocks = getBlocks();
+		const courseOutline = blocks.find(
+			( block ) => block.name === courseOutlineBlockName
+		);
+
+		if ( ! courseOutline ) {
+			return {};
+		}
+
+		for ( const moduleOrLesson of getBlocks( courseOutline.clientId ) ) {
+			if ( canPreview( moduleOrLesson ) ) {
+				return getBlockAttributes( moduleOrLesson.clientId );
+			}
+
+			if ( moduleOrLesson.name === courseModuleBlockName ) {
+				for ( const lesson of getBlocks( moduleOrLesson.clientId ) ) {
+					if ( canPreview( lesson ) ) {
+						return getBlockAttributes( lesson.clientId );
+					}
+				}
+			}
+		}
+
+		return {};
+	} );
+
+	let previewUrl = '';
+	if ( firstLesson?.id && currentPost?.id ) {
+		if (
+			globalLearningModeEnabled ||
+			currentPost.meta._course_theme === SENSEI_THEME
+		) {
+			previewUrl = `/?p=${ firstLesson.id }`;
+		} else {
+			previewUrl = `/?p=${ firstLesson.id }&learn=1&${ SENSEI_PREVIEW_QUERY }=${ currentPost.id }`;
+		}
+	}
+
+	let customizerUrl = '';
+	if ( previewUrl ) {
+		customizerUrl = `/wp-admin/customize.php?url=${ encodeURIComponent(
+			previewUrl
+		) }`;
+	}
+
 	return (
 		<PluginDocumentSettingPanel
 			name="sensei-course-theme"
-			title={ __( 'Course Theme', 'sensei-lms' ) }
+			title={ __( 'Course Styles', 'sensei-lms' ) }
 		>
-			<p>
-				{ __(
-					'This does not change the theme of your site. It only applies to logged in users when viewing the course.',
-					'sensei-lms'
-				) }
-			</p>
-			<SelectControl
-				label={ __( 'Theme', 'sensei-lms' ) }
-				hideLabelFromVision
-				value={ theme }
-				options={ [
-					{
-						label: __( 'Sensei Theme', 'sensei-lms' ),
-						value: SENSEI_THEME,
-					},
-					{
-						label: __( 'WordPress Theme', 'sensei-lms' ),
-						value: WORDPRESS_THEME,
-					},
-				] }
-				onChange={ setTheme }
-			/>
+			{ globalLearningModeEnabled ? (
+				<>
```

Comment:
I think since we just have the `<p>` element here, we don't need the Fragment, right? So we could save some indentation. =)

Reply:
Yep. Removed.